### PR TITLE
Add shipping section 

### LINKED ISSUE #2592: Implement shipping methods section in dashboard

### LINKED ISSUE #3769: Return ShippingZone after shippingPriceDelete mutation
### What I'm trying to achieve
To refresh ShippingZone's cached object in dashboard 2.0 after shipping method deletion.

### Describe a proposed solution
Return ShippingZone after `shippingPriceDelete` mutation.

### PR DESCRIPTION
**Pull request by dominik-zeglen**
I want to merge this change because it adds shipping section to the Dashboard 2.0.
Resolves #2592 
Resolves #3769

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by dominik-zeglen** [comment]
Blocked by #3769

File: saleor/static/dashboard-next/components/CardTitle/CardTitle.tsx [hunk 1]
```diff
@@ -49,13 +53,21 @@ const CardTitle = withStyles(styles, { name: "CardTitle" })(
...
     ...props
   }: CardTitleProps) => (
     <>
-      <div className={classNames(classes.root, className)} {...props}>
+      <div
+        className={classNames({
+          [classes.root]: true,
+          [className]: !!className,
+          [classes.constantHeight]: height === "const"
+        })}
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/components/CardTitle/CardTitle.tsx]: Why not

```
classNames(classes.root, {
          [className]: !!className,
          [classes.constantHeight]: height === "const"
        })}
```

File: saleor/static/dashboard-next/components/MoneyRange/MoneyRange.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}) => (
+  <LocaleConsumer>
+    {locale =>
+      from && to
+        ? i18n.t("{{ fromMoney }} - {{ toMoney }}", {
+            context: "money",
+            fromMoney: from.amount.toLocaleString(locale, {
+              currency: from.currency,
+              style: "currency"
+            }),
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/components/MoneyRange/MoneyRange.tsx]: Imo u could do sth like:

```
const extractCurreny = field =>
  field.amount.toLocaleString(locale, {
    currency: field.currency,
    style: "currency"
  });
```

File: saleor/static/dashboard-next/shipping/components/ShippingZoneCreatePage/ShippingZoneCreatePage.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                      )}
+                      disabled={disabled}
+                      emptyText={
+                        data.default
+                          ? i18n.t(
+                              "This is default shipping zone, which means that it covers all of the countries which are not assigned to other shipping zones"
+                            )
+                          : i18n.t(
+                              "Currently, there are no countries assigned to this shipping zone"
+                            )
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/shipping/components/ShippingZoneCreatePage/ShippingZoneCreatePage.tsx]:
```
i18n.t(
  data.default
    ? "This is default shipping zone, which means that it covers all of the countries which are not assigned to other shipping zones"
    : "Currently, there are no countries assigned to this shipping zone"
);
```

File: saleor/static/dashboard-next/shipping/components/ShippingZoneDetailsPage/ShippingZoneDetailsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                disabled={disabled}
+                emptyText={maybe(
+                  () =>
+                    shippingZone.default
+                      ? i18n.t(
+                          "This is default shipping zone, which means that it covers all of the countries which are not assigned to other shipping zones"
+                        )
+                      : i18n.t(
+                          "Currently, there are no countries assigned to this shipping zone"
+                        ),
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/shipping/components/ShippingZoneDetailsPage/ShippingZoneDetailsPage.tsx]: ditto

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+              };
+
+              const onShippingZoneUpdate = (data: UpdateShippingZone) => {
+                if (data.shippingZoneUpdate.errors.length === 0) {
+                  pushMessage({
+                    text: i18n.t("Successfully updated shipping zone")
+                  });
+                  closeModal();
+                }
+              };
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx]: DRY

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+                        );
+                        const deleteZoneTransitionState = getMutationState(
+                          ops.shippingZoneDelete.opts.called,
+                          ops.shippingZoneDelete.opts.loading,
+                          maybe(
+                            () =>
+                              ops.shippingZoneDelete.opts.data
+                                .shippingZoneDelete.errors
+                          )
+                        );
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx]: DRY

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx [hunk 7]
```diff
@@ -0,0 +1,513 @@
...
+                  )}
+                </ShippingZoneOperations>
+              );
+            }}
+          </Shop>
+        )}
+      </Messages>
+    )}
+  </Navigator>
+);
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx]: IMO this component is too big and not realy readable. Consider splitting/cleaning it.

File: saleor/static/dashboard-next/components/MoneyRange/MoneyRange.tsx (hunk 2 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/components/MoneyRange/MoneyRange.tsx]: Soo dry

File: saleor/static/dashboard-next/shipping/components/ShippingZoneCreatePage/ShippingZoneCreatePage.tsx (hunk 3 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/shipping/components/ShippingZoneCreatePage/ShippingZoneCreatePage.tsx]: AFAIK not gonna work with https://github.com/i18next/i18next-scanner

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx (hunk 5 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx]: TBH I don't see any clean way to simplify this.

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx (hunk 6 above)
Comment (dominik-zeglen) [reply saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx]: #3791

File: saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx (hunk 5 above)
Comment (piotrgrundas) [reply saleor/static/dashboard-next/shipping/views/ShippingZoneDetails/index.tsx]: Simple function which would accept key or `data.shippingZoneUpdate` etc

**Pull request merged**